src/components/List.js

```diff
@@ -26,10 +28,47 @@ const List = ({ token }) => {
   };
 
   const handleCheckbox = async (event) => {
-    const currentDate = Date.now(); // changed to using Date.now(), time given in milliseconds
+    const currentDate = Date.now(); // time given in milliseconds
     const queryCollection = await db.collection(token).doc(event.target.id);
+    // filteredListItem is our current snapshot of the doc
+    const filteredListItem = docs.filter(
+      (doc) => doc.id === queryCollection.id,
+    );
+    const listItem = filteredListItem[0];
+    const previouslyPurchasedDate = listItem.lastPurchased;
+    const latestInterval = currentDate - previouslyPurchasedDate;
+    // If item has not been purchased set to 1, else increment by 1
+    const numberOfPurchases = !listItem.numberOfPurchases
+      ? 1
+      : (listItem.numberOfPurchases += 1);
```

Comment:
In functional programming, which is the paradigm that React more or less ascribes to, it is thought of as best practice to avoid mutating data as it can cause non-deterministic behavior really hard to catch bugs. 

In this case, we are mutating `listItem.numberOfPurchases` with `+=`. I think that we can just replace `+=` with `+` as we're really just looking to assign the value of `listItem.numberOfPurchases + 1` to `numberOfPurchases`

Here's a good little article about immutability in functional programming - https://dev.to/mpodlasin/functional-programming-in-js-part-ii-immutability-vanilla-js-immutable-js-and-immer-2ccm

Reply:
Yes! Of course 🤦 that's React's whole thing... I think it totally escaped us that that's what we were doing there. Thank you for pointing it out.